lsp-mode.el

```diff
@@ -3103,13 +3080,12 @@ disappearing, unset all the variables related to it."
                                          (hierarchicalDocumentSymbolSupport . t)))
                       (formatting . ((dynamicRegistration . t)))
                       (rangeFormatting . ((dynamicRegistration . t)))
-                      ,@(pcase lsp-semantic-highlighting
-                          ((or :immediate :deferred) '((semanticHighlightingCapabilities . ((semanticHighlighting . t)))))
-                          (:semantic-tokens `((semanticTokens
-                                               . ((tokenModifiers . ,(if lsp-semantic-tokens-apply-modifiers
-                                                                         (apply 'vector (mapcar #'car lsp-semantic-token-modifier-faces)) []))
-                                                  (tokenTypes . ,(apply 'vector (mapcar #'car lsp-semantic-token-faces)))))))
-                          (_ '()))
+                      ,@(if lsp-enable-semantic-highlighting
```

Comment:
if -> when?

Reply:
> This is a lot of code!

Hope you mean a lot of deleted code? That's largely because I had two similar, but separate implementations for the old protocol living side by side. I'll also move the face definitions upwards so I can get rid of the "reference to free variable" warnings caused by the references to `lsp-semantic-token-modifier-faces` and `lsp-semantic-token-faces` in lines 3086 and 3087. Hope that doesn't lead to awful diffs

Reply:
> if -> when?

thanks! Changed it

Reply:
> > This is a lot of code!
> 
> Hope you mean a lot of deleted code? That's largely because I had two similar, but separate implementations for the old protocol living side by side. I'll also move the face definitions upwards so I can get rid of the "reference to free variable" warnings caused by the references to `lsp-semantic-token-modifier-faces` and `lsp-semantic-token-faces` in lines 3086 and 3087. Hope that doesn't lead to awful diffs

Yes, a lot of deleted code (like something good).

Reply:
alright, I don't need to worry then :) I'm glad we could get rid of that deadwood